pkg/api/defaults.go

```diff
@@ -508,6 +512,69 @@ func (p *Properties) setHostedMasterProfileDefaults() {
 	p.HostedMasterProfile.Subnet = DefaultKubernetesMasterSubnet
 }
 
+func (p *Properties) setCustomCloudProfileDefaults() {
+
+	if p.IsAzureStackCloud() {
+
+		//azureStackCloudSpec is the default configurations for azure stack with public Azure.
+		azureStackCloudSpec := AzureEnvironmentSpecConfig{
+			CloudName: AzureStackCloud,
+			//DockerSpecConfig specify the docker engine download repo
+			DockerSpecConfig: DefaultDockerSpecConfig,
+			//KubernetesSpecConfig is the default kubernetes container image url.
+			KubernetesSpecConfig: DefaultKubernetesSpecConfig,
+			DCOSSpecConfig:       DefaultDCOSSpecConfig,
+			EndpointConfig: AzureEndpointConfig{
+				ResourceManagerVMDNSSuffix: "",
+			},
+			OSImageConfig: map[Distro]AzureOSImageConfig{
+				Ubuntu:          DefaultUbuntuImageConfig,
+				RHEL:            DefaultRHELOSImageConfig,
+				CoreOS:          DefaultCoreOSImageConfig,
+				AKS:             DefaultAKSOSImageConfig,
+				AKSDockerEngine: DefaultAKSDockerEngineOSImageConfig,
+			},
+		}
+
+		//Set default value for ResourceManagerVMDNSSuffix
+		azureStackCloudSpec.EndpointConfig.ResourceManagerVMDNSSuffix = p.CustomCloudProfile.Environment.ResourceManagerVMDNSSuffix
+
+		// Use the custom input to overwrite the default values in AzureStackCloudSpec
+		if p.CustomCloudProfile.AzureEnvironmentSpecConfig != nil {
+			helpers.AssignStringIfNotEmpty(&p.CustomCloudProfile.AzureEnvironmentSpecConfig.CloudName, p.CustomCloudProfile.AzureEnvironmentSpecConfig.CloudName)
```

Comment:
A couple of Issues here

i) We have too many struct field retrievals. We can minimize that by assigning to variables

For eg:-  

i)  `azureStackCloudSpec.DockerSpecConfig` can be assigned to `dockerSpecConfig`
ii) `azureStackCloudSpec.KubernetesSpecConfig` can be assigned to  `kubernetesSpecConfig`
and so on.
and these vars can be reused throughout.